optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/listener/support/VariableListenerNotification.java

```diff
@@ -58,8 +59,7 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        // A direct implementation (instead of HashCodeBuilder) to avoid dependencies
-        return (((17 * 37) + System.identityHashCode(entity))) * 37 + type.hashCode();
+        return Objects.hash(entity, type);
```

Comment:
This is not a pure refactor: now the entity's hashcode is used instead of the entity's identityHashCode.
*Breaks equals contract* (and this one unlike the ones above this one will cause trouble): the equals method uses entity same semantics, but the hashcode uses entity equals semantics.
For the record, it needs to remain entity same semantics (the more we force our users to implement equals/hashcode the way we want it, the less compatible we are with other Java frameworks)
PS: we probably want a unit test proving that this breaks the equals contract before we fix it - clearly the original code is missing test coverage :)